uspace/lib/c/include/fibril_synch.h

```diff
@@ -90,10 +90,11 @@ typedef struct {
 	fibril_rwlock_t name = FIBRIL_RWLOCK_INITIALIZER(name)
 
 typedef struct {
+	long counter;
 	list_t waiters;
-} fibril_protocvar_t;
+} fibril_semaphore_t;
 
-#define FIBRIL_PROTOCVAR_INITIALIZER(name) \
+#define FIBRIL_SEMAPHORE_INITIALIZER(name) \
```

Comment:
Shouldn't we initialize the counter here as well?

Reply:
Good idea!